addons/mail/models/res_partner.py

```diff
@@ -36,7 +37,20 @@ def _compute_contact_address_inline(self):
             partner.contact_address_inline = re.sub(r'\n(\s|\n)*', ', ', partner.contact_address).strip().strip(',')
 
     def _compute_im_status(self):
-        super()._compute_im_status()
+        status_by_partner = {}
+        # sudo: partner can access their presences
+        for presence in self.user_ids.sudo().presence_id:
+            partner = presence.user_id.partner_id
+            if (
+                status_by_partner.get(partner, "offline") == "offline"
+                or presence.status == "online"
+            ):
+                status_by_partner[partner] = presence.status
```

Comment:
here also, no need to build a mapping, thanks to having `presence_id` we can just follow relations directly in code below